thread_pool.cpp

```diff
@@ -0,0 +1,121 @@
+#include "thread_pool.hpp"
+#include <iostream>
+#include <boost/asio/post.hpp>
+#include <boost/assert.hpp>
+#include <spdlog/spdlog.h>
+#include "buffer_pool.hpp"
+
+using namespace std;
+using namespace ezio;
+
+read_job::read_job(char *buffer, buffer_pool *pool,
+	std::function<void(libtorrent::disk_buffer_holder,
+		libtorrent::storage_error const &)>
+		handler) :
+	buffer_(buffer), pool_(pool), handler_(handler)
+{
+}
+
+void read_job::operator()()
+{
+	// do read/write operation.
+
+	lt::storage_error error;
+	error.operation = lt::operation_t::file_read;
+	error.ec = libtorrent::errors::no_error;
+
+	handler_(libtorrent::disk_buffer_holder(*pool_, buffer_, 123), error);
+
+	SPDLOG_INFO("buffer: {}", buffer_);
+}
+
+void write_job::operator()()
+{
+	SPDLOG_ERROR("write_job is not implemented");
+}
+
+void hash_job::operator()()
+{
+	SPDLOG_ERROR("hash_job is not implemented");
+}
+
+thread_pool *thread_pool::get_instance()
+{
+	static thread_pool inst;
+	return &inst;
+}
```

Comment:
please don't make it as singleton.
in the future, we might need several instances